Applications/Spire/Source/Ui/DecimalSpinBox.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Spire/Ui/DecimalSpinBox.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+DecimalSpinBox::DecimalSpinBox(double value, QWidget* parent)
+    : QAbstractSpinBox(parent) {
+  m_spin_box = new RealSpinBox(static_cast<long double>(value),
+    this);
+  setFocusProxy(m_spin_box);
+  m_spin_box->connect_change_signal([=] (auto value) {
+    m_change_signal(value.extract_double());
+  });
+  connect(m_spin_box, &RealSpinBox::editingFinished, this,
+    &DecimalSpinBox::on_editing_finished);
```

Comment:
Is it possible to directly connect the two?

    connect(m_spin_box, &RealSpinBox::editingFinished, this,
      &DecimalSpinBox::editingFinished);

Reply:
I can use a lambda but I'd still have to convert the values.

Reply:
Ah, nevermind. I see it now.